intercoop_addons/account_reconcile_pos_payments/models/account_bank_statement.py

```diff
@@ -80,91 +80,146 @@ def _reconcile_pos(self):
                     'The journal %s has no CB Contract number.'
                     ) % rec.journal_id.display_name)
 
+            if rec.journal_id.cb_contactless_matching and not rec.journal_id.cb_contract_number_contactless:
+                raise UserError(_(
+                    'The journal %s has no CT Contract number.'
+                ) % rec.journal_id.display_name)
```

Comment:
```suggestion
            if rec.journal_id.cb_contactless_matching and not rec.journal_id.cb_contract_number_contactless:
                raise UserError(_(
                    'The journal %s has no CT Contract number.'
                ) % rec.journal_id.display_name)
```

I think it's better to move this validation to a constraint in the `account.journal` model
If you add that constraint you make sure that the contract_number is set when contactless_matching is enabled, and you don't need to check it here

Reply:
Hi @siddharth7815 have you seen this comment?

Reply:
ok, Do you want me to fix it and create new PR for versoin 9.0 or Version 12.0?

Reply:
Change it in 12.0, this is ok for 9.0